internal/proto/reader.go

```diff
@@ -159,6 +177,9 @@ func (r *Reader) ReadReply() (interface{}, error) {
 
 	switch line[0] {
 	case RespStatus:
+		if r.opt.useStatusStringType {
```

Comment:
This fixes the tests, but do we worry it might introduce an unexpected condition when enabled?

Reply:
> do we worry it might introduce an unexpected condition when enabled?
No, i don't think so. This change just means when Teleport is using the Reader explicitly/directly, we always get the `StatusString` type, which is what the Teleport db engine wants. And when Reader is used internally by the go-redis driver itself, it's a string like the old day.